ansible/roles/test/tasks/neighbour-mac-noptf.yml

```diff
@@ -31,35 +31,28 @@
   - name: init loganalyzer for later syslog analysis
     include: roles/test/files/tools/loganalyzer/loganalyzer_init.yml
 
-  - name: gather DUT arp table
-    switch_arptable:
+  - name: gather DUT interface table
+    show_interface: command='status'
 
-  ########## Test V4 mac address change #################
-  - name: pick IPv4 neighbor to test change mac behavior
+  - name: select A routing interface for testing
     set_fact:
-      v4_nei: "{{ item.key }}"
-      v4_intf: "{{ item.value['interface'] }}"
-    with_dict: "{{ arptable.v4 }}"
+      routing_interface: "{{ item.key }}"
+    with_dict: "{{int_status}}"
     when:
-      - ('Ethernet' in item.value['interface']) or ('PortChannel' in item.value['interface'])
-      - arptable.v4 | length != 0
-
-  - block:
-    - name: select Ethernet0 if cannot find a v4 neighbor for test
-      set_fact:
-        v4_intf: "Ethernet0"
-        v4_nei: "{{ v4_intf_nei }}"
+      - ('routed' in item.value['vlan'] and 'up' in item.value['oper_state'])
```

Comment:
This line is what I was referring to 'vlan'?

Reply:
whether an interface is a routed one is displayed in the "vlan" column of the output of "show interface", which might be the term "vlan" you have seen.

Reply:
Sure. 
Can I just add a comment here and don't change the key name of "intf_status"? I prefer to have the key name aligned with the title of the column where its data come from.

Reply:
Agree Stephen. We can just add a comment to clarify this.

Reply:
It should have been 'master' in the first place instead of 'vlan' in `show int status`. At this point, a comment would suffice.